### ISSUE TITLE
getBoxProof API endpoint

### ISSUE THREAD

**Issue by greenhat** [ergoplatform/ergo#1525]
### Motivation

To simplify payment verification in SPV client workflow planned in https://github.com/ergoplatform/sigma-rust/issues/495

This will allow us to skip the tx verification step (1.2 and 2.1).

### Implementation details

Use `Header.stateRoot` AVL+ tree digest of UTXO set.

**Comment by kushti** [comment]
@greenhat it seems we can't do a proof for arbitrary header as only current state is available. Any possible workarounds around that? 

@knizhnik please do API method anyway:

* it should be located in the /utxo section
* it should be named getBoxesBinaryProof (/utxo/getBoxesBinaryProof)
* for given set of box ids, it should return hex-encoded binary-serialized batch proof

**Comment by greenhat** [comment]
> @greenhat it seems we can't do a proof for arbitrary header as only current state is available. Any possible workarounds around that?

Following our discussion on [discord](https://discord.com/channels/668903786361651200/669989266478202917/920048607242125332) where we established that we can get UTXO proof only for the current state(block). So we have to wait for n blocks to get some confirmations after we got UTXO proof.
I think in a scenario where the payment was made a few moments ago it makes perfect sense. We still have to wait for n blocks for confirmation anyway. Moreover, NiPoPoW does not help since we don't have the suffix(any blocks after the target) yet.

So in the end we will have two options to do SPV:
1. Recent payment will be verified with UTXO proof and waiting for n blocks confirmation;
2. Old payment will be verified via NiPoPoW proof of the block, then tx inclusion proof.
